### PR TITLE
Add WebHDFS based CSV downloader

### PR DESCRIPTION
**Pull request by vidma**
Works like this: 
- save CSV to hdfs, reducing to a single file named `download.csv`
- and then produce a link for downloading:
  * via WebHDFS if `WEB_HDFS_URL` environment variable is set (i.e. `WEB_HDFS_URL="http://some-company:some-port"`)
  * or simply `hdfs:///some-hdfs-path` otherwise

Would this make sense to be part of `spark-notebook` ?

some more complex alternatives or future additions could be:
- proxying hdfs file via `spark-notebook` (consumes resources right?)
- detecting `hdfs:///some-hdfs-path` link in output, and automatically formatting it as a link

@andypetrella 

screen shot 2015-12-02 at 19 15 32

### CODE REVIEW

**Comment by andypetrella** [comment]
LGTM, however I guess you want this var in order to be able to download the file in a distributed environment which is ok.

However it could be something else than HDFS, like glusterfs etc. 

Would that make sense it be more generic?

**Comment by vidma** [comment]
> Would that make sense it be more generic?

sure. it only requires some way to generate a download link, so [a little update here](https://github.com/andypetrella/spark-notebook/pull/411/files#diff-a35f3ca29824ef65c8c93013b8233ec9R56) would do.

I suggest to leave env var specific as is, and we could add more cases if needed.

**Comment by andypetrella** [comment]
yeah, let's have the feature first! like it

File: project/Shared.scala [hunk 1]
```diff
@@ -88,6 +88,7 @@ object Shared {
         sparkCore(sv),
         sparkYarn(sv),
         sparkSQL(sv),
+        "com.databricks" %% "spark-csv" % "1.3.0",
```
Comment (andypetrella) [inline project/Shared.scala]: maybe move that to a variable?

Comment (vidma) [inline project/Shared.scala]: :ok:

File: project/Dependencies.scala [hunk 2]
```diff
@@ -104,6 +103,13 @@ object Dependencies {
     ExclusionRule("org.apache.hadoop")
   ) excludeAll(parquetList:_*)
 
+  def sparkCSV: Seq[ModuleID] = {
+    import scala.math.Ordering.Implicits._
+    if (sparkVersionTuple >= (1, 3, 0)) {
```
Comment (vidma) [inline project/Dependencies.scala]: maybe this check isn't worth, do we care about spark older than `1.3` or `1.4`?

looks like `df.write.format("xyz")` was introduced in spark 1.4, so this line [might not compile against older spark](https://github.com/andypetrella/spark-notebook/pull/411/files#diff-a35f3ca29824ef65c8c93013b8233ec9R43).

Comment (andypetrella) [inline project/Dependencies.scala]: actually you put that in the post-df thingy which is after 1.3.0 so we're safe.

At some point, we'll have to stop supporting these stoneage version :-D

**Pull request merged**